hivemq-edge/src/main/java/com/hivemq/bridge/metrics/PerBridgeMetrics.java

```diff
@@ -34,54 +39,68 @@ public class PerBridgeMetrics {
+        publishForwardFailCounter = createBridgeCounter(metricRegistry,
                 bridgeName,
                 "forward.publish.failed",
-                "count"));
+                "count");
 
-        publishRemoteReceivedCounter = metricRegistry.counter(MetricRegistry.name(BRIDGE_PREFIX,
+        publishRemoteReceivedCounter = createBridgeCounter(metricRegistry,
                 bridgeName,
                 "remote.publish.received",
-                "count"));
+                "count");
 
-        publishLocalReceivedCounter = metricRegistry.counter(MetricRegistry.name(BRIDGE_PREFIX,
+        publishLocalReceivedCounter = createBridgeCounter(metricRegistry,
                 bridgeName,
                 "local.publish.received",
-                "count"));
+                "count");
 
-        publishLocalSuccessCounter =
-                metricRegistry.counter(MetricRegistry.name(BRIDGE_PREFIX, bridgeName, "local.publish", "count"));
+        publishLocalSuccessCounter = createBridgeCounter(metricRegistry,
+        bridgeName,
+                "local.publish",
+                "count");
 
-        publishLocalNoSubscriberCounter = metricRegistry.counter(MetricRegistry.name(BRIDGE_PREFIX,
+        publishLocalNoSubscriberCounter = createBridgeCounter(metricRegistry,
                 bridgeName,
                 "local.publish.no-subscriber-present",
-                "count"));
+                "count");
 
-        publishLocalFailCounter = metricRegistry.counter(MetricRegistry.name(BRIDGE_PREFIX,
+        publishLocalFailCounter = createBridgeCounter(metricRegistry,
                 bridgeName,
                 "local.publish.failed",
-                "count"));
+                "count");
 
-        remotePublishExcludedCounter = metricRegistry.counter(MetricRegistry.name(BRIDGE_PREFIX,
+        remotePublishExcludedCounter = createBridgeCounter(metricRegistry,
                 bridgeName,
                 "forward.publish.excluded",
-                "count"));
+                "count");
 
-        loopPreventionForwardDropCounter = metricRegistry.counter(MetricRegistry.name(BRIDGE_PREFIX,
+        loopPreventionForwardDropCounter = createBridgeCounter(metricRegistry,
                 bridgeName,
                 "forward.publish.loop-hops-exceeded",
-                "count"));
+                "count");
 
-        loopPreventionRemoteDropCounter = metricRegistry.counter(MetricRegistry.name(BRIDGE_PREFIX,
+        loopPreventionRemoteDropCounter = createBridgeCounter(metricRegistry,
                 bridgeName,
                 "remote.publish.loop-hops-exceeded",
-                "count"));
+                "count");
+    }
+
+    private Counter createBridgeCounter(final @NotNull MetricRegistry metricRegistry, @NotNull final String... names){
+        final String metricName = MetricRegistry.name(BRIDGE_PREFIX, names);
+        synchronized (mutex){
```

Comment:
What is the point of creating a mutex object and not synchronizing on 'this'? Would have the same semantic as the mutex is a normal field, right?

Reply:
Generally I consider synchronizing on "this" a bad pattern, it means external objects can cause your encapsulated state change to be affected by external conditions. Nothing is stopping arbitrary code sync'ing the reference and locking you up. Encapsulation of these semantics wins in highly contended threaded code